ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/api/CreatePluginUtils.kt

```diff
@@ -48,20 +48,26 @@ public fun <PluginConfigT : Any> createClientPlugin(
     name: String,
     createConfiguration: () -> PluginConfigT,
     body: ClientPluginBuilder<PluginConfigT>.() -> Unit
-): ClientPlugin<PluginConfigT> =
-    object : ClientPlugin<PluginConfigT> {
-        override val key: AttributeKey<ClientPluginInstance<PluginConfigT>> = AttributeKey(name)
```

Comment:
One of the problems was here because. AttributeKey call inside a non-inline function produces something like this:
```java
KTypeParameter var10006 = Reflection.typeParameter(new FunctionReferenceImpl(3, CreatePluginUtilsKt.class, "createClientPlugin", "createClientPlugin(Ljava/lang/String;Lkotlin/jvm/functions/Function0;Lkotlin/jvm/functions/Function1;)Lio/ktor/client/plugins/api/ClientPlugin;", 1), "PluginConfigT", KVariance.INVARIANT, false);
```
And it doesn't work after obfuscation.